### PR TITLE
task portion by adira

### PR DESCRIPTION
**Pull request by AdiraBlumenthal**
added task.py and updated stepcvt with new commands

### CODE REVIEW

File: scripts/stepcvt [hunk 1]
```diff
@@ -29,5 +29,12 @@ if __name__ == "__main__":
     epp.add_argument("-c", "--count", type=int, help="Change the count of the part")
     epp.set_defaults(func=part.edit_part)
 
+    cvtp = sp.add_parser("stlconvert", help="specify how to convert to stl")
+    cvtp.add_argument("partID", nargs="*", help="Specify partid for parts to be added")
+    cvtp.add_argument("--rotation", type=int, nargs=3, help="3 integers for x, y, z rotation")
+    cvtp.add_argument("--linearTolerance", type=float, nargs=1, help="float for linear tolerance")
```
Comment (ymh1003) [inline scripts/stepcvt:35]: `nargs=1` will produce a list which contains one item.

File: scripts/stepcvt [hunk 2]
```diff
@@ -29,5 +29,12 @@ if __name__ == "__main__":
     epp.add_argument("-c", "--count", type=int, help="Change the count of the part")
     epp.set_defaults(func=part.edit_part)
 
+    cvtp = sp.add_parser("stlconvert", help="specify how to convert to stl")
+    cvtp.add_argument("partID", nargs="*", help="Specify partid for parts to be added")
```
Comment (ymh1003) [inline scripts/stepcvt:33]: * means that the number of arguments can be zero.

File: stepcvt/cli/task.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    else:
+        source = p.sources[0]
+
+    if args.partID:
+        partID = args.partID
+    else:
+        print("ERROR: Need to specify at least one partid if --all is not included")
+        return 1
+
+    if args.rotation == True:
```
Comment (ymh1003) [inline stepcvt/cli/task.py:35]: `args.rotation` will return a list of 3 integers. It will never be True.

File: stepcvt/cli/task.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    )
+    partID.rotate(source)
+
+    with open(args.jsonfile, "w") as jf:
+        json.dump(partID.to_dict(), jf)
+    return 0
+
+
+# issues:
+# - there is no real way to connect a part to an STLConversionInfo object besides naming the object the partID
```
Comment (ymh1003) [inline stepcvt/cli/task.py:57]: I think you should first find the part in source using the `args.partid` and then either change the existing `STLConversionInfo` or create a new one.

File: stepcvt/cli/task.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    else:
+        angularTol = 0.1
+
+    partID = project.STLConversionInfo(
+        rotation=rotation, linearTolerance=linearTol, angularTolerance=angularTol
+    )
+    partID.rotate(source)
+
+    with open(args.jsonfile, "w") as jf:
+        json.dump(partID.to_dict(), jf)
```
Comment (ymh1003) [inline stepcvt/cli/task.py:52]: Should we update the jsonfile with the changed `project`? Right now the jsonfile only contains the information of a `STLConversionInfo`.

File: stepcvt/cli/task.py (hunk 4 above)
Comment (AdiraBlumenthal) [reply stepcvt/cli/task.py:57]: @ymh1003 are we storing a list of all the STLConversionInfo objects somewhere and if so, where?

Comment (ymh1003) [reply stepcvt/cli/task.py:57]: Each `partinfo` object has an attribute called `info`. This stores a list of `Taskinfo` objects, which may or may not contain the `STLConversionInfo` object.

Comment (AdiraBlumenthal) [reply stepcvt/cli/task.py:57]: And source has partinfo objects so I could do source.contains(partID) and get the corresponding partinfo object for the ID?

Comment (AdiraBlumenthal) [reply stepcvt/cli/task.py:57]: Oh I see - it is like this part of your code 
`for partid, obj in source.parts():
                    if partid in ids:
                        if partid in existing_ids:
                            print(f"{partid} has already been added")
                        else:
                            source.add_partinfo(partid, obj)`

Comment (ymh1003) [reply stepcvt/cli/task.py:57]: Right right.

**Comment by AdiraBlumenthal** [comment]
@ymh1003 I updated the pull request. Can you take a look and see if this addresses the issues?

**Pull request merged**